src/execution/compiler/operator/update_translator.cpp

```diff
@@ -213,7 +213,10 @@ void UpdateTranslator::GenTableUpdate(FunctionBuilder *builder) const {
 
   auto *cond = GetCodeGen()->UnaryOp(parsing::Token::Type::BANG, update_call);
   If success(builder, cond);
-  builder->Append(GetCodeGen()->AbortTxn(GetExecutionContext()));
+  {
```

Comment:
I don't really have time to look at this now, but I just wanted to leave a comment in case this got merged before I was able to look at it. It's possible that `analyze_translator` needs to get updated too since I based it heavily on `update_translator`

Reply:
I think I made the corresponding change.